src/cpu/core_dyn_x86/cache.h

```diff
@@ -114,6 +115,17 @@ class CodePageHandler : public PageHandler {
 		}
 		return is_current_block;
 	}
+
+	uint8_t *alloc_invalidation_map() const
+	{
+		constexpr size_t map_size = 4096;
```

Comment:
We've (annoyingly) go some duplicates here too - 
include/mem.h:#define MEM_PAGESIZE 4096
include/paging.h:#define MEM_PAGE_SIZE	(4096)

Reply:
Maybe; I considered this and perhaps it's the correct way to go, but I am not 100% sure yet - that's why I copied hardcoded value, like it was in the original code. An alternative would be to add a new define (perhaps `CODE_PAGE_SIZE` or something like that).

I am not sure if *this* 4096 is the same as DOS memory page size (which I presume is value indicated by those two headers). Perhaps they are completely independent and using 4096 was just lazy programming by an original author? Are code page sizes (and consequently "map" sizes used here) the same value? (I don't know) Maybe we can rise this code page size (to improve efficiency) while keeping DOS memory page size as it is (for compatibility)?

Aside of those 2 headers, there's also `#define PAGESIZE 4096` in `src/cpu/core_dyn_x86.cpp` and `src/cpu/core_dynrec.cpp` (and few more hardcoded "4096"s in there) (if you're wondering why those cpp files are relevant here - files in `src/cpu/core_dyn_x86/` and `src/cpu/core_dynrec/` are not "real" headers - they are implementation files included in the respective cpp files :/.

This is a fallout of bad practices used byt original authors - they used "magic values" and they did not pick descriptive names :( In result it's now harder to understand and modify this code.

Reply:
> I am not sure if this 4096 is the same as DOS memory page size (which I presume is value indicated by those two headers). Perhaps they are completely independent and using 4096 was just lazy programming by an original author?

Yes - I wish the magic numbers could have been commented at the time of implementation.  Much of the audio code is littered with 4096, 8192, 16384 buffer sizes.. even though the callbacks only ever ask for audio in 1ms chunks, so the largest butter needed is ( audio-rate Hz / 1000 s/ms ) * 2 (if stereo) * bit-depth-per-sample.

If we consider the CD-DA allows 96 kHz audio and 32bit samples, then: 96000 / 1000 * 2 * 4 = 768 bytes.  
But most audio channels top out at 48 kHz 8bit mono (48 byte buffer, like the Tandy) or 16bit-stereo (192 byte buffer, like the SB16). 

Given this 4096 is undocumented, I agree it's the safest to keep it un-linked to the other 4096 defines, as doing so could mislead future maintainers (or ourselves :-).